Source/WebCore/rendering/RenderLayerScrollableArea.cpp

```diff
@@ -2025,4 +2028,14 @@ float RenderLayerScrollableArea::deviceScaleFactor() const
     return m_layer.renderer().document().deviceScaleFactor();
 }
 
+Element* RenderLayerScrollableArea::elementForAnchoringController()
+{
+    return m_layer.renderer().element();
+}
+
+IntRect RenderLayerScrollableArea::boundingRectForAnchoringController()
+{
+    return IntRect(m_layer.renderer().absoluteBoundingBoxRect());
```

Comment:
I don't think this give the right behavior. Above you do:
```
ScrollOffset(FloatPoint(candidate.absoluteBoundingBoxRect().location() - m_owningScrollableArea.boundingRectForAnchoringController().location()));
```

but that doesn't give you a scroll offset inside the overflow:scroll.

This implies that the testing isn't detecting the bug (or maybe this just explains some failing tests).

Reply:
Yeah I don't think this offset it right for subscrollers. The spec defines the offset as: 

> the block start edge of the anchor node’s [scroll anchoring bounding rect](https://drafts.csswg.org/css-scroll-anchoring/#scroll-anchoring-bounding-rect), relative to the block start edge of the scrolling content in the [block flow direction](https://drafts.csswg.org/css-writing-modes-4/#block-flow-direction) of the scroller

not sure what the best way to implement this currently is (maybe just offsetFromContainer?).

Reply:
Updated this to `computeOffsetFromOwningScroller` to make it more obvious what this function does.